subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -424,6 +427,45 @@ public void setDebug(boolean enabled) {
         forkOptions.setDebug(enabled);
     }
 
+    /**
+     * Enables fail fast behavior causing the task will fail on the first failed test.
+     */
+    @Option(option = "fail-fast", description = "Stops test execution after the first failed test.")
+    public Test failFastOn(boolean ignored) {
+        setFailFast(true);
+        return this;
+    }
+
+    /**
+     * Disables fail fast behavior causing the task to run all tests.
+     */
+    @Option(option = "no-fail-fast", description = "Ensures test execution continues after a failed test.")
```

Comment:
❌ Not necessary, the user can use `--fail-fast=false`. If we think that `--no-fail-fast` is important, we should bake this as a general feature into our `@Option` handling, not add additional methods here.

Reply:
@bmuschko had a similar comment, but it looks like `@Option` does not support this.  When I tried running with `--fail-fast=false`, Gradle outputs:
```
$ gradle test --fail-fast=false
...
FAILURE: Build failed with an exception.

* What went wrong:
Problem configuring task :test from command line.
> Command-line option '--fail-fast' does not take an argument.
```

Reply:
@acourtneybrown I don't think the boolean option supports the behavior. You can only use `--fail-fast` (== true) or not use it at all (== false). See https://docs.gradle.org/nightly/userguide/custom_tasks.html#sec:supported_task_option_data_types.

Reply:
@bmuschko @oehme after digging a little more, it looks like `AbstractOptionElement.calculateOptionType()` is where the code explicitly disallows passing `true` or `false` values to `[bB]oolean` command line args:

```
    protected static Class<?> calculateOptionType(Class<?> type) {
        //we don't want to support "--flag true" syntax
        if (type == Boolean.class || type == Boolean.TYPE) {
            return Void.TYPE;
        } else {
            return type;
        }
    }
```

Are you okay with having the 2 command line options (`--fail-fast` & `--no-fail-fast`) being handled explicitly?  Or do we want to add handling to `@Option` to automatically create a `no` version of `[bB]oolean` command line args?

Reply:
I don't think we should change the behavior of `@Option` right now. Plugin developers were already using the internal API of `@Option`. We also just made `@Option` a public API with 4.6.

In general I am wondering how often users will want to use the `false` option. Probably almost never. A user made the conscious decision to set `fail-fast` to `true` (likely by configuring the task). I cannot think of too many reasons why you'd want to temporary revert that decision. And if a user wants to not use fail-fast (even temporarily) then you can just change the flag in the build script or make it configurable via project property or something else.

Bottom line: My vote goes to just exposing the command line option `--fail-fast` and nothing else.

Reply:
Let's spit this into 2 issues:

- Do we need the **feature** `--no-fail-fast` (or `--fail-fast false`, the name is not important)
- How do we implement it, by adding `--no-fail-fast` manually, or try something automatically?

For issue 2, I agree with @oehme , this should be generated automatically, for example, when `--fail-fast` is declared, `--no-fail-fast` is implicitly declared, we don't need to declare it manually. If so, issue1 seems to be solved as well - we don't need to declare it.

Reply:
@acourtneybrown I should have been clearer in my comment above: I think we should do *neither* in this PR. No extra option (since no other task does that) and no change to Option handling. 

We can revisit this later if anyone actually asks for it.

Reply:
How I understand @oehme's comment here is that we don't want _any_ command line option at the moment. None of the other Test configuration properties are exposed via the command line.

Reply:
And my interpretation of "no extra option" was to remove `--no-fail-fast` but keep `--fail-fast`.

@oehme can you clarify which way you want it?